packages/advanced-logic/specs/payment-network-erc20-proxy-contract-0.1.0-DRAFT.md

```diff
@@ -0,0 +1,241 @@
+| ------------------------- | -------------------------------------------------------------- |
+| **id**                    | "pn-erc20-proxy-contract"                                      |
+| **type**                  | "paymentNetwork"                                               |
+| **version**               | "0.1.0"                                                        |
+| **values**                |                                                                |
+| **values.paymentAddress** | `paymentAddress` from parameters if given, undefined otherwise |
+| **values.refundAddress**  | `refundAddress` from parameters if given, undefined otherwise  |
+| **values.salt**           | Salt for the request                                           |
+| **events**                | Array with one 'create' event (see below)                      |
+
+the 'create' event:
+
+|  Property                     |  Value                                                         |
+| ----------------------------- | -------------------------------------------------------------- |
+| **name**                      | 'create'                                                       |
+| **parameters**                |                                                                |
+| **parameters.paymentAddress** | `paymentAddress` from parameters if given, undefined otherwise |
+| **parameters.refundAddress**  | `refundAddress` from parameters if given, undefined otherwise  |
+| **parameters.salt**           | Salt for the request                                           |
+
+---
+
+### Updates
+
+#### addPaymentAddress
+
+##### Parameters
+
+|                               | Type   | Description                               | Requirement   |
+| ----------------------------- | ------ | -----------------------------------       | ------------- |
+| **id**                        | String | Constant value: "pn-erc20-proxy-contract" | **Mandatory** |
+| **action**                    | String | Constant value: "addPaymentAddress"       | **Mandatory** |
+| **parameters**                | Object |                                           |               |
+| **parameters.paymentAddress** | String | Ethereum address for the payment          | **Mandatory** |
+
+##### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc20-proxy-contract" exists
+- The signer is the `payee`
+- The extension property `paymentAddress` is undefined
+
+##### Warnings
+
+None.
+
+##### Results
+
+An extension state is updated with the following properties:
+
+|  Property                  |  Value                                               |
+| -------------------------- | ---------------------------------------------------- |
+| **values.paymentAddress**  | `paymentAddress` from parameters                     |
+| **parameters.paymentSalt** | Salt for the payment                                 |
```

Comment:
`| **parameters.paymentSalt** | Salt for the payment                                 |`
line to delete